generators/spring-boot/command.ts

```diff
@@ -73,12 +75,12 @@ const command = {
         when: () => ['gateway', 'microservice'].includes(gen.jhipsterConfigWithDefaults.applicationType),
         type: 'list',
         message: 'Which service discovery server do you want to use?',
-        default: 'consul',
+        default: CONSUL,
       }),
       choices: [
-        { value: 'consul', name: 'Consul (recommended)' },
-        { value: 'eureka', name: 'JHipster Registry (legacy, uses Eureka, provides Spring Cloud Config support)' },
-        { value: 'no', name: 'No service discovery' },
+        { value: CONSUL, name: 'Consul (recommended)' },
```

Comment:
This kind of change can loose typescript derived property inference.

Reply:
So true, thanks for the remark, I'll rollback this part